x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -1086,16 +1088,36 @@ const updateExistingDataStream = async ({
     }
   }
 
+  const filterDimensionMappings = (templates?: Array<Record<string, MappingDynamicTemplate>>) =>
+    templates?.filter(
+      (template) => (Object.values(template)[0].mapping as any)?.time_series_dimension
+    ) ?? [];
+
+  const currentDynamicDimensionMappings = filterDimensionMappings(
+    currentBackingIndexConfig?.mappings?.dynamic_templates
+  );
+  const updatedDynamicDimensionMappings = filterDimensionMappings(mappings.dynamic_templates);
+
+  const dynamicDimensionMappingsChanged =
```

Comment:
It is possible that some order are different between the two array? can we sort those array?  also I have a small preference to use something like `fast-deep-equal` here as it seems more robust

Reply:
thanks for the suggestion, updated